generator/internal/discovery_type_vertex.h

```diff
@@ -44,16 +44,16 @@ class DiscoveryTypeVertex {
   std::string const& name() const { return name_; }
   std::string const& package_name() const { return package_name_; }
   nlohmann::json const& json() const { return json_; }
+  std::set<DiscoveryTypeVertex*> needs_type() { return needs_type_; }
+  std::set<DiscoveryTypeVertex*> needed_by_type() { return needed_by_type_; }
```

Comment:
Given the look of the other accessors, it seems like these should probably return by reference too, and particularly so given that there is an ownership question about the set members.